### PR TITLE
Add SuperTuxKart egg

### PR DESCRIPTION
**Pull request by mattman107**
# Description

This pull request adds an egg file for the SuperTuxKart server.

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/SuperTuxKart/egg-super-tux-kart.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "meta": {
+        "version": "PTDL_v2",
+        "update_url": null
+    },
+    "exported_at": "2024-03-23T21:35:53-04:00",
+    "name": "SuperTuxKart",
+    "author": "mattamn107@github.com",
+    "description": "Egg for hosting a SuperTuxKart Server. \r\nhttps:\/\/supertuxkart.net\/Main_Page\r\n\r\nFirst install will take a while because the games source code and assets will be downloaded and the games executable will have to be compiled.\r\n\r\nTo change server settings, stop the server then edit the 'config.xml' file in the root of the container. When done save the file and start the server.\r\n\r\nTo update the server (only should have to be done when the games code or assets are updated) in panel go to Servers > SuperTuxKart Server > Manage > Reinstall. This will download the updated code\/assets (should exclude anything that didn't change) and recompile the executable.\r\n\r\nCurrently this is setup to run a local lan server. To make this a wan server (available in the SuperTuxKart online lobby list) read the document at this link and make the necessary changes to the startup command: https:\/\/github.com\/supertuxkart\/stk-code\/blob\/master\/NETWORKING.md",
```
Comment (QuintenQVD0) [inline game_eggs/SuperTuxKart/egg-super-tux-kart.json:10]: This is way to long. And should just be a short description of the game.

File: game_eggs/SuperTuxKart/egg-super-tux-kart.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            "container": "ghcr.io\/parkervcp\/installers:debian",
+            "entrypoint": "bash"
+        }
+    },
+    "variables": []
```
Comment (QuintenQVD0) [inline game_eggs/SuperTuxKart/egg-super-tux-kart.json:30]: Their no variables? To specify other version or change things in the config file?

File: game_eggs/SuperTuxKart/ReadMe.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# SuperTuxKart
+
+SuperTuxKart is a 3D open-source arcade racer with a variety characters, tracks, and modes to play.
+
+## Files downloaded from their [Github](https://github.com/supertuxkart/stk-code) and [SourceForge](https://svn.code.sf.net/p/supertuxkart/code/stk-assets)
```
Comment (QuintenQVD0) [inline game_eggs/SuperTuxKart/ReadMe.md:5]: This is not needed. This is not usefull for the end user.

File: game_eggs/SuperTuxKart/ReadMe.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+|---------|---------|
+| Game    | 2759   |
+
+### Notes
+
+To change server settings, stop the server then edit the 'config.xml' file in the root of the container. When done save the file and start the server.
+
+2759 is the default port, but any port can be used. You will have to manually change it in the config.xml file.
+
+To update the server (only should have to be done when the games code or assets are updated) in panel go to Servers > SuperTuxKart Server > Manage > Reinstall. This will download the updated code/assets (should exclude anything that didn't change) and recompile the executable.
```
Comment (QuintenQVD0) [inline game_eggs/SuperTuxKart/ReadMe.md:35]: This should mention it on the client side node the admin side.

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/SuperTuxKart/egg-super-tux-kart.json (hunk 2 above)
Comment (mattman107) [reply game_eggs/SuperTuxKart/egg-super-tux-kart.json:30]: Using the pterodactyl file parser I was not able to edit the config.xml file. It seems like the parser would make the changes, but then the server exe would notice that the format wasn't identical to what it was expecting and change it back.

I could add variables to change a couple of things in the start command like the name of the server and the name of the config file it should use.

File: game_eggs/SuperTuxKart/ReadMe.md (hunk 4 above)
Comment (mattman107) [reply game_eggs/SuperTuxKart/ReadMe.md:31]: I'm sorry I am confused with what you mean by this. What do you mean?

File: game_eggs/SuperTuxKart/egg-super-tux-kart.json (hunk 2 above)
Comment (QuintenQVD0) [reply game_eggs/SuperTuxKart/egg-super-tux-kart.json:30]: if it is possible as an argument to the binary then I would like that, if not then it is fine.

File: game_eggs/SuperTuxKart/ReadMe.md (hunk 4 above)
Comment (QuintenQVD0) [reply game_eggs/SuperTuxKart/ReadMe.md:31]: You mention reinstalling from the admin side, if you go on the client side, settings then their is the same reinstall button but for clients.

Comment (mattman107) [reply game_eggs/SuperTuxKart/ReadMe.md:31]: Oh okay I think I understand. I will provide instructions for both. Sorry I just kind of use pterodactyl to host my own game servers so If I have to do something I just usually go to the admin settings to do it. I haven't had the experience of a non admin user before.

File: game_eggs/SuperTuxKart/ReadMe.md [hunk 5]
```diff
@@ -28,6 +28,6 @@ To change server settings, stop the server then edit the 'config.xml' file in th
 
 2759 is the default port, but any port can be used. You will have to manually change it in the config.xml file.
 
-To update the server (only should have to be done when the games code or assets are updated) in panel go to Servers > SuperTuxKart Server > Manage > Reinstall. This will download the updated code/assets (should exclude anything that didn't change) and recompile the executable.
+To update the server (only should have to be done when the games code or assets are updated) in panel as an admin go to Servers > SuperTuxKart Server > Manage > Reinstall. If you are a standard user go to SuperTuxKart Server > Settings > Reinstall Sever. This will download the updated code/assets (should exclude anything that didn't change) and recompile the executable.
```
Comment (QuintenQVD0) [inline game_eggs/SuperTuxKart/ReadMe.md:31]: you typed sever and not server

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/SuperTuxKart/egg-super-tux-kart.json (hunk 2 above)
Comment (mattman107) [reply game_eggs/SuperTuxKart/egg-super-tux-kart.json:30]: I didn't realize it but there are actually a lot more arguments to the binary than I first thought. When I get a chance I will add some and variables to control them.

File: game_eggs/SuperTuxKart/ReadMe.md (hunk 5 above)
Comment (mattman107) [reply game_eggs/SuperTuxKart/ReadMe.md]: Updated to correct spelling.

File: game_eggs/SuperTuxKart/egg-super-tux-kart.json (hunk 2 above)
Comment (mattman107) [reply game_eggs/SuperTuxKart/egg-super-tux-kart.json]: I have added a basic set of variables to pass as arguments to the binary in the start command.

Comment (QuintenQVD0) [reply game_eggs/SuperTuxKart/egg-super-tux-kart.json]: thank you!

**Comment by mattman107** [comment]
Is there anything else that needs to be addressed for this?

**Comment by QuintenQVD0** [comment]
> Is there anything else that needs to be addressed for this?

No. I just need to find time to review it.

**Pull request merged**